lib/Controller/RoomController.php

```diff
@@ -661,6 +674,13 @@ protected function formatRoomV2andV3(Room $room, ?Participant $currentParticipan
 				$roomData['canDeleteConversation'] = $room->getType() !== Room::ONE_TO_ONE_CALL
 					&& $currentParticipant->hasModeratorPermissions(false);
 				$roomData['canLeaveConversation'] = true;
+				if ($this->getAPIVersion() >= 3) {
+					$roomData['canEnableSIP'] =
+						$this->talkConfig->isSIPConfigured()
+						&& ($room->getType() === Room::GROUP_CALL || $room->getType() === Room::PUBLIC_CALL)
+						&& $currentParticipant->hasModeratorPermissions(false)
+						&& $this->talkConfig->canUserEnableSIP($currentUser);
```

Comment:
Should it be also checked if the room token only contains digits (as otherwise [`Room::setSIPEnabled` would fail](https://github.com/nextcloud/spreed/blob/6996121c23afc68850d4e83c8f7fdae587dccecb/lib/Room.php#L746))? Or that check is just an extra safety measure and configuring SIP automatically updates the previous room tokens, so if SIP is configured they will always be only digits?

Reply:
yeah right, let me check that

Reply:
> configuring SIP automatically updates the previous room tokens,

No way, as the token is used everywhere across clients and server to store information.